### PR TITLE
Move source to prettier formatting

### LINKED ISSUE #13: Add prettier to the project
Prettier formatting is controversial but I believe is one of the most "agreeable" formattings out there, and that the whole of Facebook's source has been moved to it is a strong point.

Besides, formatting will always be a case of taste and that will defer from person to person. This usually leads to conflicts between people in a code review (comments like "put a space after `if`" and another "don't put a space after `if`").

Arguing about formatting in a code review takes the focus away from the actual code, and to how the code looks to one person.

Adding prettier as a linting rules takes the formatting variable out the code review. Everyone, whether they like it or not, will follow the same rule.

### PR DESCRIPTION
**Pull request by squgeim**
- Add npm script to run prettier for js files in src and test.
- Add prettier as an eslint rule

Fixes #13

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -14,6 +14,7 @@
     "migrate": "knex migrate:latest --knexfile src/knexfile.js",
     "rollback": "knex migrate:rollback --knexfile src/knexfile.js",
     "test": "NODE_ENV=test mocha --recursive --require babel-register test",
+    "prettier": "prettier --single-quote --trailing-comma es5 --write \"src/**/*.js\" \"test/**/*.js\"",
```
Comment (mesaugat) [inline package.json]: I would not want a trailing comma, it looks alien to me even though it has it's benefits. And can you also increase the print width to 120 characters and re-run prettier again?

Comment (mesaugat) [reply package.json]: Rename the key to `prettify` as well.

Comment (mesaugat) [reply package.json]: We need to make an update here too: https://github.com/mesaugat/express-api-es6-starter/blob/master/package.json#L20-L25

Comment (squgeim) [reply package.json]: 👍 on 120 lines and no trailing comma (only just got comfortable to both and prefer them, but I agree it looks awkward to most).

I don't think we need to change the `lint-staged` config. I have already added the `prettier` plugin to eslint. `prettier` is run as an eslint rule and can be applied as `eslint --fix`.

Comment (mesaugat) [reply package.json]: Great 👍

**Pull request merged**